### DISCUSSION TITLE
How to change menu background color?

### DISCUSSION THREAD

**Discussion by TicTak21** [A-l-y-l-e/Alyle-UI#172] [Q&A] [answered]
I have a following template:

```html
<button
  color="primary:contrast"
  ly-button
  appearance="icon"
  [lyMenuTriggerFor]="menu"
>
  <ly-icon
    >notifications<ly-badge bg="accent">{{
      countNotifications
    }}</ly-badge></ly-icon
  >
</button>

<ng-template #menu let-menuRef>
  <ly-menu [ref]="menuRef">
    <a
      [routerLink]="['/notifications', item.id]"
      ly-button
      ly-menu-item
      *ngFor="let item of notifications"
    >
      {{ item.title }}
    </a>
  </ly-menu>
</ng-template>
```

And I can't change menu background color.

## What I already tried:

1) Inline styling
  ```html
  <ly-menu [ref]="menuRef" [style.background]="'red'">...</ly-menu>
  ```
Result: strange bug with background

https://user-images.githubusercontent.com/44474697/154453387-043c5e04-144d-455f-906f-e98586ae7fdd.mov

2) Ly-paper(as wrapper or attribute)
Result: nothing changed

3) Global styling via `GlobalVariables`
```ts
export class CustomGlobalVariables implements PartialThemeVariables {
  menu = {
    root: () => lyl`{
      background: red
    }`,
  };
}
```

Result: same as (1)👆

## Solution(?)
There is a `<div>` inside menu:

<img width="938" alt="Screenshot 2022-02-17 at 12 12 08" src="https://user-images.githubusercontent.com/44474697/154454360-d4e726d6-8023-4d82-adee-a36e3ef9e63e.png">

and if I change div background via devtools I get a desired background:
<img width="111" alt="Screenshot 2022-02-17 at 12 13 59" src="https://user-images.githubusercontent.com/44474697/154454708-4f0c3e33-1083-4083-82f0-ee9761223ca3.png">

**Comment by Enlcxx** [comment, marked as answer]
Hi, sorry for the delay. There are several ways to do it.

## 1. Global styling via `GlobalVariables` or a specific theme
```ts
export class CustomGlobalVariables implements PartialThemeVariables {
  menu = {
    root: (__) => lyl `{
      ${__.container} {
        background: blue
      }
    }`,
  };
}
```

## 2. Global styling via `$global`
Adding this style to `AppComponent`:
```ts
...
import { StyleRenderer, lyl, ThemeVariables, ThemeRef } from '@alyle/ui';
import { STYLES as MENU_STYLES } from '@alyle/ui/menu';

const STYLES = (theme: ThemeVariables, ref: ThemeRef) => {
  ref.renderStyleSheet(MENU_STYLES);
  const menu = ref.selectorsOf(MENU_STYLES);
  return {
    $global: lyl `{
      body {
        background-color: ${theme.background.default}
        color: ${theme.text.default}
        font-family: ${theme.typography.fontFamily}
        margin: 0
        direction: ${theme.direction}
      }
      ${menu.container} { <------------- add this 
        background: red
      }
    }`,
    ...
  }
}
```
## 3. Style for specific menu
[Stackblitz demo](https://stackblitz.com/edit/angular-r1zivk?file=app%2Fmenu-demo-01.component.ts)
`some-component.ts`
```ts
const STYLES = (_theme: ThemeVariables, ref: ThemeRef) => {
  ref.renderStyleSheet(MENU_STYLES);
  const menu = ref.selectorsOf(MENU_STYLES);
  return {
    menuCustom: () => lyl`{
      ${menu.container} {
        background: red
      }
    }`,
  };
};
```
```html
<ly-menu [ref]="M" [ngClass]="classes.menuCustom">
```

**Comment by TicTak21** [comment]
Thank you!
